proposals/XXXX-efficient-token-program.md

```diff
@@ -0,0 +1,97 @@
+---
+simd: 'XXXX'
+title: 'p-token: Efficient Token program'
+authors:
+  - febo (Anza)
+  - Jon Cinque (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2025-03-19
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+Replace the current version of SPL Token (`TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA`) program by a CU-optimized one (`p-token`).
+
+## Motivation
+
+About `~10%` of block compute units is used by the Token program instructions. Decreasing the CUs used by Token program instructions creates block space for other transactions to be executed – i.e., less CUs consumed by Token program, more CUs for other transactions.
+
+As an example, if we reduce the CUs consumed by Token instructions to `1/20th` of their current value, `10%` of block CUs utilized becomes `0.5%`, resulting in a `9.5%` block space gain.
+
+Additionally, there are benefits to downstream programs:
+* Better composability since using the Token program instructions require less CUs.
+* Cheaper (in terms of CU) cross-program invocations.
+
+## New Terminology
+
+N/A.
+
+## Detailed Design
+
+`p-token` is a like-for-like efficient re-implementation of the Token program. It is `no_std` (no heap memory allocations are made in the program) and uses zero-copy access for instruction and account data. Since it follows the same instructions and accounts layout, it does not require any changes to client code – it works as a drop-in replacement.
+
+Apart from the original SPL Token instructions, this proposal adds two additional instructions to the program:
+
+1. [`withdraw_excess_lamports`](https://github.com/solana-program/token/blob/main/p-token/src/processor/withdraw_excess_lamports.rs) (instruction discriminator `38`): allow recovering "bricked" SOL from mint (e.g., USDC mint as `~323` SOL in excess) and multisig accounts. The logic of this instruction is the same as the current SPL Token-2022 instruction. 
```

Comment:
It's probably best to call out that there could be some economic consequence of unbricking some SOL, and that we haven't calculated how much extra SOL could be freed up in this way.